### PR TITLE
Support more sophisticated cells in RenderTable

### PR DESCRIPTION
**Pull request by rleibman**
Allows more sophisticated cell rendening in RenderTable. Before, each column could only refer to an item in the map, now the renderer function gets passed the full object to be rendered as well as the key to the column that we're rendering. This allows more interesting renderings, for example, a column with buttons that can be pressed to take action on the contents of the row.

### CODE REVIEW

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala [hunk 1]
```diff
@@ -10,18 +10,22 @@ import scalacss.ScalaCssReact._
...
+  type CellRenderer = (Model, ColumnKey) ⇒ ReactElement
 
-  /**
-   *  ._1: String = column name
-   *  ._2: Option[Any => ReactElement] = custom cell
-   *  ._3: Option[(Model,Model) => Boolean] = sorting function
-   *  ._4: Option[Double] = column width interms of flex property
-   */
-  type Config = (String, Option[Any => ReactElement], Option[(Model, Model) => Boolean],Option[Double])
+  case class ColumnConfig(name: String,
```
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:17]: Good idea - i hated that type alias since the first time i saw it :)
Can we move it out side of `ReactTable`, so you dont need to prefix or import it to use it?

**Comment by oyvindberg** [comment]
Can you please remove all checked in `.cache-main` files , remove all except the one `.gitignore` which is in the root of the project, and add the eclipse specific files to that?

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala [hunk 2]
```diff
@@ -10,18 +10,22 @@ import scalacss.ScalaCssReact._
 
 object ReactTable {
 
-  type Model = Map[String, Any]
+  type ColumnKey = String
+  type Model = Map[ColumnKey, Any]
```
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:14]: My biggest issue with this component is this `Any` here. We could make the whole `ReactTable` a trait parameterised over a `T`, so you could have a typesafe `object ReactTableString extends ReactTable[String]...` or something like that. it's not important for now, but we can experiment with it now if you want to?

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala [hunk 3]
```diff
@@ -126,35 +120,30 @@ object ReactTable {
...
           S.offset,
           onNextClick,
-          onPreviousClick
-        )
-      )
+          onPreviousClick))
   }
 
-  def getIntSort(key: String) = (m1: Model, m2: Model) => m1(key).asInstanceOf[Int] < m2(key).asInstanceOf[Int]
+  def getIntSort(key: String) = (m1: Model, m2: Model) ⇒ m1(key).asInstanceOf[Int] < m2(key).asInstanceOf[Int]
```
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:126]: I took out all the unicode arrows (even though i like them myself) because people complained about them, and in any case we should be consistent, so we dont pollute diffs for all future. can we go with `=>` for now?

**Comment by oyvindberg** [comment]
Thanks for the contribution, i'm pretty enthusiastic about making the pure scala components better. they should be able to act like a showcase of what we can do easily, and they arent really ATM

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 1 above)
Comment (rleibman) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:17]: I'm not sure, I think it's more idiomatic scala to leave it in, particularly because this is on the "companion" object. ColumnConfig has no real meaning outside of ReactTable.

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 2 above)
Comment (rleibman) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:14]: I know! I tried, but you can't parametrize a companion object, you can parametrize everything, but that makes the file nearly illegible. I tried making it into a class, but didn't think of a trait, that may work.

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 3 above)
Comment (rleibman) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:126]: It's not me, it's eclipse, it's doing it all by itself. How are you removing them? What's the complaint?  It's 2016, who's not using a unicode compliant editor?

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 2 above)
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:14]: I actually did this when i tried to use `ReactTable` for a project, so it works. the only thing you need to specify for a given type `T` is a function `T => String` or something similar, it might have been for the rendering i needed it - dont remember well.

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 3 above)
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:126]: Yeah, i know. Anyways, we could change _all_ the arrows, but if we do, lets do it in another PR

File: core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala (hunk 1 above)
Comment (oyvindberg) [inline core/src/main/scala/chandu0101/scalajs/react/components/ReactTable.scala:17]: Fair enough, i just thought it looked a bit clumsy in the demos, but we can leave it like it is

**Comment by intracer** [comment]
It would be nice to have support for inline editing. I think merging this pull request can be a starting point for this?

**Comment by oyvindberg** [comment]
I'll close this for now. Feel free to reopen
